assets/js/view.coffee

```diff
@@ -303,17 +311,115 @@ renderLine = (lineData, options = {}) ->
           @messageDiv.text('')
         ), options.time
 
+
     ##########
-    # export
+    # EXPORT
     ##########
 
-    export: (filename, mimetype) ->
+    exportFile: (filename, mimetype) ->
       filename ||= @settings?.getSetting?('export_filename') || 'vimflowy.json'
       if not mimetype? # Infer mimetype from file extension
           mimetype = @mimetypeLookup filename
       content = @exportContent mimetype
       @saveFile filename, mimetype, content
-
+    importFileSelector: (success, failure) ->
+      # Success callback takes (content, mimetype)
+      # Failure callback takes ()
+      # Both are optional.
+      window.location.hash = "import-file"
+      resolve = () ->
+          window.onhashchange = null
+          $("#import-file .submit").off 'click'
+          window.location.hash = ""
+      if "onhashchange" in window
+          window.onhashchange = () ->
+              if failure? then do failure
+              do resolve
+      $("#import-file .submit").on 'click', () =>
+          # if file is actually uploaded
+          file = $("#import-file :file")[0].files[0]
+          if file?
+              mimetype = @mimetypeLookup file.name
+              reader = new FileReader()
+              reader.readAsText file, "UTF-8"
+              reader.onload = (evt) ->
+                  content = evt.target.result
+                  if success? then success(content, mimetype)
+                  do resolve
+              reader.onerror = (evt) ->
+                  if failure? then do failure
+                  do resolve
+    importFile: () ->
+      @importFileSelector (content, mimetype) =>
+          @importContent content, mimetype
+    parseJson: (content) ->
+      try
+        root = JSON.parse(content)
+      catch
+        throw "Tried to import malformed JSON"
+      verify = (node) ->
+        unless node.text || node.text == '' then return false
+        if node.children
+          for child in node.children
+            unless verify child then return false
+        return true
+      unless verify root
+        throw "Tried to import JSON, but the JSON in not in vimflowy's import format"
+      return root
```

Comment:
nit: not a big deal but i prefer a space between method names (and even between long blocks of code)